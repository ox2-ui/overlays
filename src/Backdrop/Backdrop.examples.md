### Backdrop

    <Backdrop zDepth={2} isOpen={false} />
