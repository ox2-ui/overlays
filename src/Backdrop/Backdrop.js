import React, { PropTypes } from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import styled from 'styled-components';

const Base = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  -webkit-backdrop-filter: blur(7px);
  background-color: rgba(94, 90, 89, 0.6);
  animation-duration: 1s;
  z-index: -1;
`;

/**
 * Backdrop Component
 */
const Backdrop = (props) => {
  const { isOpen, onClick } = props;

  return (
    <ReactCSSTransitionGroup
      transitionName="transition:overlay-fadeIn"
      transitionAppear={true}
      transitionEnter={false}
      transitionAppearTimeout={700}
      transitionLeaveTimeout={400}
    >
      {isOpen &&
        <Base
          key={1}
          onTouchMove={event => event.preventDefault()}
          onClick={onClick}
        />
      }
    </ReactCSSTransitionGroup>
  );
};

Backdrop.propTypes = {
  /**
   * Controls whether the backdrop is visible or not.
   */
  isOpen: PropTypes.bool.isRequired,
  /**
   * Callback function that is fired when the overlay clicked.
   */
  onClick: PropTypes.func,
};

export default Backdrop;