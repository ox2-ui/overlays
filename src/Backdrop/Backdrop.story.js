import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Backdrop from './Backdrop';

storiesOf('Backdrop', module)
  .add('opened', () => (
    <Backdrop zDepth={2} isOpen={true} />
  ))
  .add('closing', () => (
    <Backdrop zDepth={2} isOpen={false} />
  ));
