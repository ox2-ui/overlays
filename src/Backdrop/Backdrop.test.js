/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Backdrop from './Backdrop';

it('renders enabled', () => {
  const wrapper = mount(
    <Backdrop zDepth={2} isOpen={true} />
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders disabled', () => {
  const wrapper = mount(
    <Backdrop zDepth={2} isOpen={false} />
  );

  expect(wrapper).toMatchSnapshot();
});
