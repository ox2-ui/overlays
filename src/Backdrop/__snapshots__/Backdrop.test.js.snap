exports[`test renders disabled 1`] = `
<Backdrop
  isOpen={false}
  zDepth={2}>
  <ReactCSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={700}
    transitionEnter={false}
    transitionLeave={true}
    transitionLeaveTimeout={400}
    transitionName="transition:overlay-fadeIn">
    <ReactTransitionGroup
      childFactory={[Function]}
      component="span"
      transitionAppear={true}
      transitionAppearTimeout={700}
      transitionEnter={false}
      transitionLeave={true}
      transitionLeaveTimeout={400}
      transitionName="transition:overlay-fadeIn">
      <span />
    </ReactTransitionGroup>
  </ReactCSSTransitionGroup>
</Backdrop>
`;

exports[`test renders enabled 1`] = `
<Backdrop
  isOpen={true}
  zDepth={2}>
  <ReactCSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={700}
    transitionEnter={false}
    transitionLeave={true}
    transitionLeaveTimeout={400}
    transitionName="transition:overlay-fadeIn">
    <ReactTransitionGroup
      childFactory={[Function]}
      component="span"
      transitionAppear={true}
      transitionAppearTimeout={700}
      transitionEnter={false}
      transitionLeave={true}
      transitionLeaveTimeout={400}
      transitionName="transition:overlay-fadeIn">
      <span>
        <ReactCSSTransitionGroupChild
          appear={true}
          appearTimeout={700}
          enter={false}
          leave={true}
          leaveTimeout={400}
          name="transition:overlay-fadeIn">
          <styled.div
            onTouchMove={[Function]}>
            <div
              className="kQeSPN"
              onTouchMove={[Function]} />
          </styled.div>
        </ReactCSSTransitionGroupChild>
      </span>
    </ReactTransitionGroup>
  </ReactCSSTransitionGroup>
</Backdrop>
`;
